src/interpreter/ByteCodeGenerator.h

```diff
@@ -196,11 +215,9 @@ struct ByteCodeGenerateContext {
 
     size_t getRegister()
     {
-        if ((m_baseRegisterCount + 1) > REGULAR_REGISTER_LIMIT) {
+        if (UNLIKELY(m_baseRegisterCount >= REGULAR_REGISTER_LIMIT)) {
             throw "register limit exceed while generate byte code";
         }
-        RELEASE_ASSERT(m_baseRegisterCount + 1 < REGULAR_REGISTER_LIMIT);
```

Comment:
I'm not sure that this check code(`RELEASE_ASSERT`) is necessary as the above throw statement already check the condition.
so I removed it.

Reply:
Nice catch!